prediction-polls/backend/src/repositories/ProfileDB.js

```diff
@@ -282,5 +282,30 @@ async function followerProfiles(userId) {
         return { error: errorCodes.DATABASE_ERROR };
     }
 }
+async function getRankPerTag(topic){
+    try {
+        const query = 'SELECT user.id , user.username has_domain_point.amount from user, has_domain_point where user.id = has_domain_point.userID AND has_domain_point.topic = ? Order By has_domain_point.amount Desc LIMIT 100';
+        const [result] = await pool.query(query, [topic]);
+        return {result: result};
+    } catch (error) {
+        return { error: errorCodes.DATABASE_ERROR };
+    }
+}
+
+async function updateBadges(userId){
+    try {
+        const query = 'SELECT userId, topic, RANK() OVER (PARTITION BY topic ORDER BY amount DESC) as userRank FROM has_domain_point where userRank > 4';
```

Comment:
This sql also has some problems. Please fix it also

Reply:
Select * from (SELECT userId, topic, RANK() OVER (PARTITION BY topic ORDER BY amount DESC) as userRank FROM has_domain_point) as temp where temp.userRank > 4
![image](https://github.com/bounswe/bounswe2023group4/assets/52269552/bcbef2ea-84e8-411d-a917-eb119670bb1d)